packages/typescript-estree/src/create-program/getParsedConfigFile.ts

```diff
@@ -0,0 +1,65 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import type * as ts from 'typescript/lib/tsserverlibrary';
+
+import { CORE_COMPILER_OPTIONS } from './shared';
+
+/**
+ * Utility offered by parser to help consumers parse a config file.
+ *
+ * @param configFile the path to the tsconfig.json file, relative to `projectDirectory`
+ * @param projectDirectory the project directory to use as the CWD, defaults to `process.cwd()`
+ */
+function getParsedConfigFile(
+  tsserver: typeof ts,
+  configFile: string,
+  projectDirectory?: string,
+): ts.ParsedCommandLine | string {
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  if (tsserver.sys === undefined) {
+    throw new Error(
+      '`getParsedConfigFile` is only supported in a Node-like environment.',
+    );
+  }
+
+  const getCurrentDirectory = (): string =>
+    (projectDirectory && path.resolve(projectDirectory)) || process.cwd();
```

Comment:
Nitpick - avoid nested logical short-circuiting as it can be hard to read and reason about.

```suggestion
    projectDirectory ? path.resolve(projectDirectory) : process.cwd();
```